supervisor/supervisor.mk

```
@@ -33,6 +33,8 @@ endif
 # (Right now INTERNAL_FLASH_FILESYSTEM and (Q)SPI_FLASH_FILESYSTEM are mutually exclusive.
 # But that might not be true in the future.)
 ifdef EXTERNAL_FLASH_DEVICES
+  COMMA := ,
+  EXTERNAL_FLASH_DEVICE_COUNT := $(words $(subst $(COMMA), ,$(EXTERNAL_FLASH_DEVICES)))
```

I was worried about whitespace after commas, so I tried a bunch of variations. `"A, B"` counts as two, that's fine, but leading or trailing whitespace will add to the count, so adding a `$strip()` will help. However, removing the quotes is tricky. https://stackoverflow.com/a/10430975/142996
```suggestion
  EXTERNAL_FLASH_DEVICE_COUNT := $(words $(subst $(COMMA), ,$(strip  $(subst $\",,$(EXTERNAL_FLASH_DEVICES)))))
```

Test `Makefile`:
```make
COMMA := ,

TEST = "  A,B "
COUNT := $(words $(subst $(COMMA), ,$(strip $(subst $\",,$(TEST)))))

all:
	echo $(TEST) $(COUNT)
```
Without the `$(strip` and `$subst`, `"  A,B " counts as 4 instead of 2.

---

I took a different approach in mk2, counting the array size in C instead...